adapters_test.go

```diff
@@ -0,0 +1,355 @@
+	}{
+		{
+			name:     "converts single deposit tx without signer or from address",
+			depTxNum: 1,
+		},
+		{
+			name:        "converts one deposit and one non-deposit tx without signer or from address",
+			depTxNum:    1,
+			nonDepTxNum: 1,
+		},
+		{
+			name:        "converts multiple deposit and non-deposit txs without signer or from address",
+			depTxNum:    10,
+			nonDepTxNum: 10,
+		},
+		{
+			name:        "converts multiple txs with from address but without signer",
+			depTxNum:    3,
+			nonDepTxNum: 3,
+			from:        "from",
+		},
+		{
+			name:        "converts multiple txs with signer but without from address",
+			depTxNum:    3,
+			nonDepTxNum: 3,
+			signTx:      noopSigner,
+		},
+		{
+			name:        "converts multiple txs with both from address and signer",
+			depTxNum:    3,
+			nonDepTxNum: 3,
+			from:        "from",
+			signTx:      noopSigner,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			txBytes, _, hexTxBytes := generateTxsAndTxsBytes(t, test.depTxNum, test.nonDepTxNum)
+			txs, err := monomer.AdaptPayloadTxsToCosmosTxs(hexTxBytes, test.signTx, test.from)
+			require.NoError(t, err)
+
+			msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{
+				TxBytes:     txBytes[:test.depTxNum],
+				FromAddress: test.from,
+			})
+			require.NoError(t, err)
+
+			sdkTx := sdktx.Tx{
+				Body: &sdktx.TxBody{
+					Messages: []*codectypes.Any{msgAny},
+				},
+			}
+
+			if test.signTx != nil {
+				err := test.signTx(&sdkTx)
+				require.NoError(t, err)
+			}
+
+			depositSDKMsgBytes, err := sdkTx.Marshal()
+			require.NoError(t, err)
+
+			cosmosTxs := make(bfttypes.Txs, 0, 1+test.nonDepTxNum)
+			cosmosTxs = append(cosmosTxs, depositSDKMsgBytes)
+
+			for _, cosmosTx := range txBytes[test.depTxNum:] {
+				var tx ethtypes.Transaction
+				err := tx.UnmarshalBinary(cosmosTx)
+				require.NoError(t, err)
+				cosmosTxs = append(cosmosTxs, tx.Data())
+			}
+
+			require.Equal(t, cosmosTxs, txs)
+		})
+	}
+
+	t.Run("returns error when input tx contains invalid binary data", func(t *testing.T) {
+		_, err := monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{hexutil.Bytes("invalid")}, nil, "")
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when no deposit txs are present", func(t *testing.T) {
+		_, _, cosmosEthTx := testutils.GenerateEthTxs(t)
+		cosmosEthTxBytes, err := cosmosEthTx.MarshalBinary()
+		require.NoError(t, err)
+		_, err = monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{cosmosEthTxBytes}, nil, "")
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when signing tx fails", func(t *testing.T) {
+		_, depositTx, _ := testutils.GenerateEthTxs(t)
+		depositTxBytes, err := depositTx.MarshalBinary()
+		require.NoError(t, err)
+
+		_, err = monomer.AdaptPayloadTxsToCosmosTxs(
+			[]hexutil.Bytes{hexutil.Bytes(depositTxBytes)},
+			func(tx *sdktx.Tx) error {
+				return errors.New("sign tx error")
+			},
+			"")
```

Comment:
nit: we don't use this styling elsewhere in the codebase. Prefer something like the following:
```suggestion
		_, err = monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{hexutil.Bytes(depositTxBytes)}, func(_ *sdktx.Tx) error {
				return errors.New("sign tx error")
			}, "")
```